flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/batch/compact/BatchCompactOperator.java

```diff
@@ -0,0 +1,114 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.file.table.batch.compact;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.connector.file.table.CompactFileUtils;
+import org.apache.flink.connector.file.table.stream.compact.CompactMessages.CompactOutput;
+import org.apache.flink.connector.file.table.stream.compact.CompactMessages.CompactionUnit;
+import org.apache.flink.connector.file.table.stream.compact.CompactMessages.CoordinatorOutput;
+import org.apache.flink.connector.file.table.stream.compact.CompactReader;
+import org.apache.flink.connector.file.table.stream.compact.CompactWriter;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.runtime.state.StateInitializationContext;
+import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
+import org.apache.flink.streaming.api.operators.BoundedOneInput;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.util.function.SupplierWithException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * CompactOperator for compaction in batch mode. It will compact files to a target file and then
+ * emit the compacted file's path to downstream operator. The main logic is similar to {@link
+ * org.apache.flink.connector.file.table.stream.compact.CompactOperator} but skip some unnecessary
+ * operations in batch mode.
+ */
+public class BatchCompactOperator<T> extends AbstractStreamOperator<CompactOutput>
+        implements OneInputStreamOperator<CoordinatorOutput, CompactOutput>, BoundedOneInput {
+
+    private static final long serialVersionUID = 1L;
+
+    private final SupplierWithException<FileSystem, IOException> fsFactory;
+    private final CompactReader.Factory<T> readerFactory;
+    private final CompactWriter.Factory<T> writerFactory;
+
+    private transient FileSystem fileSystem;
+    private transient Map<String, List<Path>> compactedFiles;
+
+    public BatchCompactOperator(
+            SupplierWithException<FileSystem, IOException> fsFactory,
+            CompactReader.Factory<T> readerFactory,
+            CompactWriter.Factory<T> writerFactory) {
+        this.fsFactory = fsFactory;
+        this.readerFactory = readerFactory;
+        this.writerFactory = writerFactory;
+    }
+
+    @Override
+    public void initializeState(StateInitializationContext context) throws Exception {
```

Comment:
will this `initializeState`  be invoked by streaming job? BTW, I think this should be put in `open()`

Reply:
Yes. I think put it in `initializeState` is more consistent to existing code. e.g, CompactCoordinator also init in `initializeState` method.